---
title: "ChIP Sequencing"
tags:
    - omics
    - NGS
    - epigenetics
date: "2024-10-30"
thumbnail: "/assets/img/thumbnail/sample.png"
bookmark: true
---

# ChIPÏù¥ÎûÄ?
---

**Ch**romatin **I**mmuno**P**recipitation followed by sequencing (ChIP-Seq)ÏúºÎ°ú NGS Í∏∞Ïà†ÏùÑ ÌôúÏö©Ìïú ÏÑ∏Ìè¨ÎÇ¥ Ï†ÑÏÇ¨Ï°∞Ï†àÏù∏Ïûê Í∑úÎ™ÖÏùÑ ÏúÑÌï¥ Ïù¥Ïö©ÎêúÎã§. ÌäπÏ†ï Îã®Î∞±ÏßàÏùò ÏÑ∏Ìè¨ÎÇ¥ Ï†ÑÏûê Ï°∞Ï†àÏù∏ÏûêÎ°úÏÑúÏùò Í∏∞Îä•ÏùÑ ÏïåÍ≥†Ïûê Ìï†Îïå ÌäπÏ†ï Îã®Î∞±ÏßàÏù¥ bindingÌïòÎäî DNA ÏÑúÏó¥ÏùÑ NGSÎ°ú ÏãúÌÄÄÏã±ÌïòÏó¨ Ïú†Ï†ÑÏ≤¥ Ï†ÑÏ≤¥ÏóêÏÑú binding Í∞ÄÎä•Ìïú motif Î•º search ÌïúÎã§.

![ChIP1](https://media.cellsignal.com/www/images/resources/applications/chip-educational/chip-protocol-steps.png)
![ChIP2](https://www.merckmillipore.com/INTERSHOP/static/WFS/Merck-KR-Site/-/Merck/ko_KR/Freestyle/BI-Bioscience/Genomic-Analysis/Epigenetics/chip-chip-workflow.jpg)

DNAÏóê bindingÌïú protein ÌòπÏùÄ histone modificationÏùÑ Í≤ÄÏ∂úÌïòÍ∏∞ ÏúÑÌïú Î∂ÑÏÑùÎ∞©Î≤ï

---

# ChIP Seq Í∞ÑÎûµÌïú ÏÜåÍ∞ú
---

- ChIP SeqÏùÄ Îã§ÏùåÍ≥º Í∞ôÏùÄ Îã®Í≥ÑÎ°ú ÏßÑÌñâ
1. (Cross-linking) Tissue ÌòπÏùÄ cellÏóêÏÑú formaldehydeÎ•º Ïù¥Ïö©ÌïòÏó¨ target proteinÍ≥º DNAÍ∞Ñ Í≤∞Ìï©ÏùÑ Í≥†Ï†ïÏãúÌÇ®Îã§.
2. Genomic DNAÎ•º Ï∂îÏ∂úÌïú ÌõÑ, sonicationÎì±Ïùò Î∞©Î≤ïÏùÑ Ïù¥Ïö©Ìï¥ Ï∂îÏ∂úÌïú DNAÎ•º Ï°∞Í∞ÅÎÇ∏Îã§.
3. (IP)Target proteinÏóê specificÌïú Ìï≠Ï≤¥Î•º Î∂ôÏó¨ DNAÏôÄ proteinÏù¥ Í≤∞Ìï©Ìïú Î∂ÄÏúÑ, proteinÏù¥ Í≤∞Ìï©ÎêòÏßÄ ÏïäÏùÄ DNAÎ•º Î∂ÑÎ¶¨ÌïúÎã§.
4. (Reverse-cross linking) Cross-linkingÏùÑ ÌíÄÍ≥† DNAÎ•º Ï†ïÏ†úÌïòÏó¨ proteinÏù¥ Í≤∞Ìï©ÎêòÏñ¥ÏûàÎçò Î∂ÄÏúÑÏùò ÏãúÌÄÄÏã±ÏùÑ ÏßÑÌñâÌïúÎã§.

- ChIP-seq Îç∞Ïù¥ÌÑ∞Î•º Î∂ÑÏÑùÌïòÎäî Î∞©Î≤ï
1. Quality control : FastQCÎì±Ïùò Ìà¥ÏùÑ ÏÇ¨Ïö©ÌïòÏó¨ ÏãúÌÄÄÏã± Î¶¨ÎìúÎì§Ïùò qualityÎ•º ÌôïÏù∏ÌïòÍ≥†, TrimGalore, TrimmomaticÎì±ÏùÑ Ïù¥Ïö©ÌïòÏó¨ ÏãúÌÄÄÏã±Ïóê ÏÇ¨Ïö©Îêú adapterÎì§Í≥º ÏãúÌÄÄÏã± qualityÍ∞Ä ÎÇÆÏùÄ ÏÑúÏó¥Îì§ÏùÑ Ï†úÍ±∞ÌïúÎã§
2. Alignment : Bowtie, BWAÎì±Ïùò Ìà¥ÏùÑ ÏÇ¨Ïö©ÌïòÏó¨ reference genomeÏóê mappingÏùÑ ÏßÑÌñâÌïúÎã§. Ïù¥ Îïå, Ï§ëÎ≥µÎêú Î¶¨ÎìúÏù∏ PCR duplicateÏôÄ mapping qualityÍ∞Ä ÎÇÆÏùÄ ÏÑúÏó¥Îì§ÏùÑ Ï†úÍ±∞Ìï¥Ï£ºÎäî Í≥ºÏ†ïÏù¥ ÌïÑÏöîÌïòÎã§
3. Peak calling : MACS, homer, SICERÎì±Ïùò Ìà¥ÏùÑ ÏÇ¨Ïö©ÌïòÏó¨ Îã®Î∞±ÏßàÏù¥ Í≤∞Ìï©Îêú DNAÏòÅÏó≠Ïù∏ peakÎ•º ÏãùÎ≥ÑÌïúÎã§
4. Downstream analysis : Peak calling Ïù¥ÌõÑ, annotation, clustering, motif analysisÎì±Ïùò Î∂ÑÏÑùÏùÑ ÏßÑÌñâÌïúÎã§. 
    - AnnotationÏùÄ peakÍ≥º Í¥ÄÎ†®Îêú Ïú†Ï†ÑÏûê Ï†ïÎ≥¥, enhancer Ï†ïÎ≥¥ Îì±ÏùÑ Î∂ÑÏÑùÌïúÎã§. 
    - Motif analysisÎäî peak ÎÇ¥Ïùò DNAÏÑúÏó¥ÏùÑ Ïù¥Ïö©ÌïòÏó¨ ÌäπÏù¥Ï†ÅÏù∏ motif Îì§ÏùÑ Ï∞æÏïÑÎÇ∏Îã§.

# ChIP Seq Î∂ÑÏÑù
---

### ÏãúÏûëÌïòÍ∏∞ Ï†Ñ Ïö©Ïñ¥ Ï†ïÎ¶¨
$threads -> threads
$genome -> reference fasta file(genome index)
$fq -> fastq file
$align_out -> unsorted.sam
$align_bam -> unsorted.bam
$align_sorted -> sorted.bam
$align_filtered -> rmdup.bam



### Trimming(QC)
[TrimGalore](https://github.com/FelixKrueger/TrimGalore)

```
$ trim_galore -j 4 \ #ÏÇ¨Ïö©Ìï† Ïì∞Î†àÎìú Ïàò ÏßÄÏ†ï(hotspot = 4)
              --fastqc \ #quality check
              -o(--output_dir) /path/to/TrimGalore/output/directory \
              /path/to/forward/reads \
              /path/to/reverse/reads #if single end, only forward reads
```

### Mapping
[Bowtie2](https://bowtie-bio.sourceforge.net/bowtie2/index.shtml)

> Reference Genome ÌïÑÏöî!
>
> fasta(ÏóºÍ∏∞ÏÑúÏó¥) + gtf(Ïú†Ï†ÑÏûê Ï£ºÏÑùÏ†ïÎ≥¥)
[Reference Genome data Download](https://www.gencodegenes.org/)

> bowtie2-Build

```
$ bowtie2-build Desktop/ChIP-Seq/fasta/GRCh38.p13.genome.fa GRCh38
```

> bowtie2

```
bowtie2 -p $threads -q -x $genome -U $fq -S $align_out
```

example
```
$ bowtie2 -p 8 -x Desktop/ChIP-Seq/bowtie2/reference/GRCh38 file.fastq > file.sam
```

[samtools](https://www.htslib.org/)

```
$ samtools view -h -S -b -@ $threads -o $align_bam $align_out
```
example
```
$ samtools view -b -S file.sam > file.bam
```

### Sorting
[sambamba](https://github.com/biod/sambamba)
> Sort BAM file by genomic coordinates

```
sambamba sort -p -t $threads -o $align_sorted $align_bam
```
example
```
$ sambamba sort -p -t 8 -o file.sorted.bam file.bam
```

> Filter out duplicates

```
$ sambamba view -p -h -t $threads -f bam -F "[XS] == null and not unmapped and not duplicate" $align_sorted > $align_filtered
```

> Index BAM

```
$ samtools index file.rmdup.bam
```

### Peak calling
[HOMER](http://homer.ucsd.edu/homer/)

> [ÏÑ§Ïπò](https://www.notion.so/ChIP-Seq-7521d94f358d46b3a92a378b54edf18f?pvs=4#dad827534ef44fb6b7807f1557fdfa9b)

```
$ perl configureHomer.pl -install
```

> Creating tag directory

```
$ makeTagDirectory ${tagDirectory} ${filteredBAM}
```

> Peak calling - [HOMER](http://homer.ucsd.edu/homer/), [BEDtools](https://bedtools.readthedocs.io/en/latest/)

```
$ findPeaks ${tagDirectory} -style factor -o auto -i ${control_tagDirectory}
$ pos2bed.pl ${tagDirectory}/peaks.txt > ${output}.bed
```

> ReplicatesÍ∞Ä Ï°¥Ïû¨Ìï† Í≤ΩÏö∞ Í∞ÅÍ∞ÅÏùò tag directoryÏôÄ peak callingÏùÑ ÏßÑÌñâ, BEDtoolsÎ•º Ïù¥Ïö©ÌïòÏó¨ Ìï©ÏπòÍ∏∞

```
$ findPeaks ${tagDirectory_rep1} -style factor -o auto -i ${control_tagDirectory_rep1}
$ findPeaks ${tagDirectory_rep2} -style factor -o auto -i ${control_tagDirectory_rep2}
$ pos2bed.pl ${tagDirectory_rep1}/peaks.txt > ${rep1}.bed
$ pos2bed.pl ${tagDirectory_rep2}/peaks.txt > ${rep2}.bed
$ bedtools intersect -a ${rep1}.bed -b ${rep2}.bed > ${common}.bed
```


### Downstream Analysis
Peak visualization, Motif analysis, Annotating peaks, Peak clustering, Gene Ontology(GO), Heatmap and density plot

> Peak visualization - [deepTools](https://deeptools.readthedocs.io/en/develop/)

```
$ bamCoverage -b ${filteredBAM} -o ${output}.bw --binSize 20
```

> Yongha Kim is the best developer in the world.
>
> Factos üëçüëÄ

# Lists
You can organize items into ordered and unordered lists.

### Ordered Lists
To create an ordered list, add line items with numbers followed by periods. The numbers don‚Äôt have to be in numerical order, but the list should start with the number one.

```
1. First item
2. Second item
3. Third item
4. Fourth item
```

1. First item
2. Second item
3. Third item
4. Fourth item

### Unordered Lists
To create an unordered list, add dashes (`-`), asterisks (`*`), or plus signs (`+`) in front of line items. Indent one or more items to create a nested list.

```
* First item
* Second item
* Third item
* Fourth item
```

* First item
* Second item
* Third item
* Fourth item

# Code
To denote a word or phrase as code, enclose it in backticks (`).

|Markdown|HTML|Rendered Output|
|---|---|---|
|At the command prompt, type \`nano\`.|At the command prompt, type \<code>nano\</code>.|At the command prompt, type `nano`.|

### Escaping Backticks
If the word or phrase you want to denote as code includes one or more backticks, you can escape it by enclosing the word or phrase in double backticks (``).

|Markdown|HTML|Rendered Output|
|---|---|---|
|\`\`Use \`code\` in your Markdown file.\`\`|\<code>Use \`code\` in your Markdown file.\</code>|``Use `code` in your Markdown file.``|

### Code Blocks
To create code blocks that spans multiple lines of code, set the text inside three or more backquotes ( ``` ) or tildes ( ~~~ ).

```xml
<html>
  <head>
  </head>
</html>
```

```python
def foo():
  a = 1
  for i in [1,2,3]:
    a += i
```

# Horizontal Rules
To create a horizontal rule, use three or more asterisks (`***`), dashes (`---`), or underscores (`___`) on a line by themselves.

```text
***

---

_________________
```

# Links
To create a link, enclose the link text in brackets (e.g., [Blue Archive]) and then follow it immediately with the URL in parentheses (e.g., (https://bluearchive.nexon.com)).

```text
My favorite mobile game is [Blue Archive](https://bluearchive.nexon.com).
```

The rendered output looks like this:

My favorite mobile game is [Blue Archive](https://bluearchive.nexon.com).


### Adding Titles

You can optionally add a title for a link. This will appear as a tooltip when the user hovers over the link. To add a title, enclose it in quotation marks after the URL.

```text
My favorite mobile game is [Blue Archive](https://bluearchive.nexon.com "All senseis are welcome!").
```

The rendered output looks like this:

My favorite mobile game is [Blue Archive](https://bluearchive.nexon.com "All senseis are welcome!").


### URLs and Email Addresses
To quickly turn a URL or email address into a link, enclose it in angle brackets.

```text
<https://www.youtube.com/>
<fake@example.com>
```

The rendered output looks like this:
<https://www.youtube.com/>
<fake@example.com>

# Images
To add an image, add an exclamation mark (`!`), followed by alt text in brackets, and the path or URL to the image asset in parentheses. You can optionally add a title in quotation marks after the path or URL.

```text
![Tropical Paradise](/assets/img/example.jpg "Maldives, in October")
```

The rendered output looks like this: ![Tropical Paradise](/assets/img/example.jpg "Maldives, in October")

### Linking Images
To add a link to an image, enclose the Markdown for the image in brackets, and then add the link in parentheses.

```text
[![La Mancha](/assets/img/La-Mancha.jpg "La Mancha: Spain, Don Quixote")](https://www.britannica.com/place/La-Mancha)
```

The rendered output looks like this: [![La Mancha](/assets/img/La-Mancha.jpg "La Mancha: Spain, Don Quixote")](https://www.britannica.com/place/La-Mancha)

# Escaping Characters
To display a literal character that would otherwise be used to format text in a Markdown document, add a backslash (`\`) in front of the character.

```text
\* Without the backslash, this would be a bullet in an unordered list.
```

The rendered output looks like this:

\* Without the backslash, this would be a bullet in an unordered list.

### Characters You Can Escape
You can use a backslash to escape the following characters.

|Character|Name|
|---|---|
|\`|backtick|
|\*|asterisk|
|\_|underscore|
|\{\}|curly braces|
|\[\]|brackets|
|\<\>|angle brackets|
|\(\)|parentheses|
|\#|pound sign|
|\+|plus sign|
|\-|minus sign \(hyphen\)|
|\.|dot|
|\!|exclamation mark|
|\||pipe|

# HTML
Many Markdown applications allow you to use `HTML` tags in Markdown-formatted text. This is helpful if you prefer certain `HTML` tags to Markdown syntax. For example, some people find it easier to use `HTML` tags for images. Using `HTML` is also helpful when you need to change the attributes of an element, like specifying the color of text or changing the width of an image.

To use `HTML`, place the tags in the text of your Markdown-formatted file.

```
This **word** is bold. This <span style="font-style: italic;">word</span> is italic.
```

The rendered output looks like this:

This **word** is bold. This <span style="font-style: italic;">word</span> is italic.
